torch/_inductor/lowering.py

```diff
@@ -3473,22 +3473,22 @@ def pooling_size(x, i, kernel_size, stride, padding, ceil_mode):
     return x_out, ceil_mode
 
 
-fallback_max_pool2d_with_indices = fallback_handler(
-    aten.max_pool2d_with_indices.default,
-    add_to_fallback_set=False,
-)
+def should_fallback_max_pool2d_with_indices(kernel_size, dilation):
```

Comment:
`kw_const = ops.constant(kernel_width, torch.int32)` should be fine since `kernel_width` is at most 25. 

`hbase = ops.index_expr(bh * stride[0] - padding[0], torch.int32)` would in theory be a problem except that in practice `ops.index_expr` respects the kernel indexing dtype, or at least this is true for triton. That said, for the same reason we should be able to use `torch.int64` here without loss in perf.